### ISSUE TITLE
False positive with R503 inside with clause

### ISSUE THREAD

**Issue by thehesiod** [afonasev/flake8-return#4]
* Date you used flake8-return: 8/16/2019
* flake8-return version used, if any: 1.1.0
* Python version, if any: 3.6.6
* Operating System: OSX

### Description

With a function like the following:
```python
async def function():
    async with thing as foo:
        return foo.bar
```

This is valid because there can be no other return than the one inside the with context.  If enter/exit raise then nothing after it will run.  The one way this is not true is if you capture the exeption with a try/except around the async with clause.

**Comment by afonasev** [comment]
I'll try to fix it soon, thx
